From 2ba3190ab0597bd9e74c9e34fb4f084d17f35bb5 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 06 May 2019 18:49:06 +0300
Subject: [PATCH] [SW-0] pll: fix timeout check

The tout variable gets decremented after the while loop check, and
therefore on a real timeout, the tout variable will have negative
value (-1).
Change the timeout check to compare with less than 0..

Change-Id: Id3504ac1ee6487ebe6644219396a42946b23584f
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 214842d..cf622ae 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -128,7 +128,7 @@
 	while (!hl_pll_is_locked(cfg) && tout--)
 		hl_pll_udelay(1);
 
-	if (!tout) {
+	if (tout < 0) {
 		preboot_puts_raw("PLL: Timeout waiting for ");
 		preboot_puts_raw(cfg->name);
 		preboot_puts_raw(" PLL lock...\n\r");
